core/src/main/scala/org/apache/pekko/persistence/jdbc/state/scaladsl/JdbcDurableStateStore.scala

```diff
@@ -117,7 +117,7 @@ class JdbcDurableStateStore[A](
     db.run(queries.deleteFromDb(persistenceId).map(_ => Done))
 
   def deleteObject(persistenceId: String, revision: Long): Future[Done] =
-    db.run(queries.deleteFromDb(persistenceId).map(_ => Done))
+    db.run(queries.deleteBasedOnPersistenceIdAndRevision(persistenceId, revision).map(_ => Done))
```

Comment:
If the deletion operation is not allowed, there will be no warnings and exceptions. I think we should notify users in some way, such as returning Future[Boolean]

Reply:
changing a public API method/function signature is not allowed - we would need a new API

Reply:
this API overrides a super class functions so it can't change

Reply:
> changing a public API method/function signature is not allowed - we would need a new API

We have already broken ABI because of Slick 3.5.0 and this project doesn't guarantee SemVer so its a stretch to also break source compatibility however in this case it seems like we can avoid it

Reply:
this function is part of the shared Persistence API - ie it overrides a function in a super class

Reply:
Better we add a new API for this , which returns the detailed result.

Reply:
That can be done in another PR.

Reply:
I don't understand - the point is we have a common Persistence API - why would we hack in extra methods in just this implementation? Should we update the common Persistence API and uptake it in all the persistence implementations?

Reply:
I created https://github.com/apache/pekko/issues/1232 - is 1232 worth delaying the 1.1.0-M1 release for? I don't think so, myself.

Reply:
> If the deletion operation is not allowed, there will be no warnings and exceptions. I think we should notify users in some way, such as returning Future[Boolean]

Isn't a better way that doesn't break binary compatibility to just return a `Future.failed` with some sealed exception trait so users can get more information?

Reply:
can this be done in another PR? - it does not just affect the function that I am working on - it affects a number of other functions

It also looks like a change that we would want to replicate across all our persistence implementations. Some of them might already have Future failure results for failures to delete but it's quite likely that they all quietly ignore the fact that delete call deleted nothing.

I really think this is a set of changes in behaviour that should be discussed publicly before we make them.

Reply:
if DAO doesn't delete anything, throwing an exception is reasonable for me.